dashboard/config/scripts/csppd3_u4_welcome.external

```diff
@@ -1,4 +1,4 @@
-name ’CSPPD3-u4 Welcome'
```

Comment:
The character on the left is U+2019 ("Right Single Quotation Mark") and the character on the right is U+0027 ("Apostrophe") http://www.unicode.org/Public/UCA/4.0.0/allkeys-4.0.0.txt

It looks like this won't be able to parse correctly with any utf8mb4 encoding

Reply:
ok. thanks for trying. if we go with the find/replace option, we may also need to take some steps to prevent new instances of these from getting added to level dsl via the level editor on levelbuilder.

Reply:
Yep, reverted to keep `to_asciii` (which is what Suresh recommended anyways) and it works. So I suppose the question is: do we want the collation to apply to all columns or only to properties? Then I'll need to figure out another alternative for this part

Reply:
so it seems like removing `to_ascii` is the culprit, not changing the db collation... right?

Reply:
assuming `to_ascii` is the problem, it seems like we'll need do some work if we want to be able to use emoji in External levels. ideas on the table:
1. one-time search and replace to convert all existing apostrophe's to `'`. if we do this, we'll probably also want to make sure we give a friendly error message when level editors try to enter special apostrophes in the future
2. replace the `to_ascii` call with a regex-replace to convert any special apostrophe characters to `'`

Reply:
Yes! And I'm thinking I should do the migration first anyways, separate from other changes. The question is if we change the character set of just the properties column, or the whole table.

Pro of the whole table: `name` could have unicode, which might happen if we start needing to create scripts in different languages?
Pro of only the column: We don't need anything more for now, so keep it simple

Reply:
One of the gems we use ('native' maybe?) has a `to_unicode` function that might give us what we want. I like your regex-replace idea––can fix problems in the future. Although I was also running into a space character causing an error, which might be harder to fix with a regex-replace because we do need spaces some times ...

Reply:
oh I see -- I misunderstood your earlier comments. sorry about that.

my concern with only doing the one column is that you'll get inconsistent results between:
* existing databases. these will only have utf8mb4 on the properties column
* new databases. (new dev environments, your old dev environment after you do db:reset). won't these end up with all columns on utf8mb4? I believe this because there does not seem to be any per-column indication in schema.rb.

based on that, I would suggest converting all columns. If you like the idea of just one column, I'd be happy to help evaluate whether my theory here is correct -- I have a virtual machine I could test it out on without blocking my main dev environment.

Reply:
The diff of the schema file is below: it looks like it both sets the character set and collation of the table, and then just the collation of the column?

![image](https://user-images.githubusercontent.com/9142121/152447962-7c4879bd-5e1e-4abe-bc77-d3f10a796f03.png)

According to https://dev.mysql.com/doc/refman/8.0/en/charset-column.html, "If COLLATE collation_name is specified without CHARACTER SET, the character set associated with collation_name and collation collation_name are used." So I think we're good. But I'd be happy to go back to doing it to the table––that does seem more consistent with what's in the code base.

Reply:
I see what you're saying -- I didn't see a per-column annotation earlier. not sure what changed, but the diff in this screenshot looks good to me!

given that my concern turned out to be a non-issue, I think it's fine to do the minimum needed to meet your goal, i.e. just convert the one column.

Reply:
either way is fine though. whatever's easier :-)

Reply:
Okay, will merge this in. Given that of the 9 commits in this PR, 4 of them were to nullify another 4 (so only 1 commit actually being used), should I clean up the commit history before merging? Not sure what best practice/norm is here ...

Reply:
hi, yeah, at that point I would normally clean them up via `git rebase staging` or sometimes `git rebase -i staging`. thanks!